dotnet-roslyn/analyzers/Audacia.CodeAnalysis.Analyzers/src/Audacia.CodeAnalysis.Analyzers/Rules/MagicNumber/MagicNumberAnalyzer.cs

```diff
@@ -36,22 +44,91 @@ private static void AnalyzeVariable(SyntaxNodeAnalysisContext context)
                 return;
             }
 
-            foreach (var variable in variableDeclaration.Declaration.Variables)
+            AnalyzeVariables(variableDeclaration.Declaration.Variables, context, SyntaxKind.LocalDeclarationStatement);
+        }
+
+        private static void AnalyzeVariables(SeparatedSyntaxList<VariableDeclaratorSyntax> variables, SyntaxNodeAnalysisContext context, SyntaxKind kind)
+        {
+            SyntaxKind[] TypesToCheckNumericDeclarations = new SyntaxKind[] { SyntaxKind.ForStatement, SyntaxKind.IfStatement, SyntaxKind.WhileStatement };
+            foreach (var variable in variables)
             {
                 var initializer = variable.Initializer;
                 if (initializer == null)
                 {
                     return;
                 }
 
+                if (TypesToCheckNumericDeclarations.Any(type => type == kind) && initializer.Value is LiteralExpressionSyntax numericLiteralExpression)
+                {
+                    AnalyzeNumericLiteral(numericLiteralExpression, context, variable);
+                }
+
                 if (initializer.Value is BinaryExpressionSyntax binaryExpression)
                 {
-                    AnalyzeBinaryExpression(context, variable, binaryExpression);
+                    AnalyzeBinaryExpression(context, binaryExpression,variable);
                 }
             }
         }
 
-        private static void AnalyzeBinaryExpression(SyntaxNodeAnalysisContext context, VariableDeclaratorSyntax variable, BinaryExpressionSyntax binaryExpression)
+        private static void AnalyzeForSyntax(SyntaxNodeAnalysisContext context)
+        {
+            var variableDeclaration = (ForStatementSyntax)context.Node;
```

Comment:
not sure if `variableDeclaration` is the best name for this `var`? same for the other methods